nifi-nar-bundles/nifi-poi-bundle/nifi-poi-processors/src/main/java/org/apache/nifi/processors/poi/ConvertExcelToCSVProcessor.java

```diff
@@ -0,0 +1,409 @@
+
+        FlowFile ff = session.create();
+        try {
+
+            XMLReader parser =
+                    XMLReaderFactory.createXMLReader(
+                            "org.apache.xerces.parsers.SAXParser"
+                    );
+            ExcelSheetRowHandler handler = new ExcelSheetRowHandler(sst);
+            parser.setContentHandler(handler);
+
+            ff = session.write(ff, new OutputStreamCallback() {
+                @Override
+                public void process(OutputStream out) throws IOException {
+                    InputSource sheetSource = new InputSource(sheetInputStream);
+                    try {
+                        ExcelSheetRowHandler eh = (ExcelSheetRowHandler) parser.getContentHandler();
+                        eh.setFlowFileOutputStream(out);
+                        parser.setContentHandler(eh);
+                        parser.parse(sheetSource);
+                        sheetInputStream.close();
+                    } catch (Exception ex) {
+                        getLogger().error(ex.getMessage());
+                    }
+                }
+            });
+
+            if (handler.getSheetName().equals(UNKNOWN_SHEET_NAME)) {
+                //Used the named parsed from the handler. This logic is only here because IF the handler does find a value that should take precedence.
+                ff = session.putAttribute(ff, SHEET_NAME, sName);
+            } else {
+                ff = session.putAttribute(ff, SHEET_NAME, handler.getSheetName());
+                sName = handler.getSheetName();
+            }
+
+            ff = session.putAttribute(ff, ROW_NUM, new Long(handler.getRowCount()).toString());
+
+            if (StringUtils.isNotEmpty(originalParentFF.getAttribute(CoreAttributes.FILENAME.key()))) {
+                ff = session.putAttribute(ff, SOURCE_FILE_NAME, originalParentFF.getAttribute(CoreAttributes.FILENAME.key()));
+            } else {
+                ff = session.putAttribute(ff, SOURCE_FILE_NAME, UNKNOWN_SHEET_NAME);
+            }
+
+            //Update the CoreAttributes.FILENAME to have the .csv extension now. Also update MIME.TYPE
+            ff = session.putAttribute(ff, CoreAttributes.FILENAME.key(), updateFilenameToCSVExtension(ff.getAttribute(CoreAttributes.UUID.key()),
+                    ff.getAttribute(CoreAttributes.FILENAME.key()), sName));
+            ff = session.putAttribute(ff, CoreAttributes.MIME_TYPE.key(), CSV_MIME_TYPE);
+
+            session.transfer(ff, SUCCESS);
+
+        } catch (SAXException saxE) {
+            getLogger().error(saxE.getMessage());
+            ff = session.putAttribute(ff,
+                    ConvertExcelToCSVProcessor.class.getName() + ".error", saxE.getMessage());
+            session.transfer(ff, FAILURE);
+        } finally {
+            try {
+                sheetInputStream.close();
+            } catch (IOException ioe) {
+                //nothing further can be done...
```

Comment:
Does it make sense to catch it here if we can't do anything?  How about not catching the exception and letting the handlers in onTrigger get it?